lib/jxl/dec_frame.cc

```diff
@@ -737,6 +740,31 @@ Status FrameDecoder::ProcessSections(const SectionInfo* sections, size_t num,
   }
   if (has_error) return JXL_FAILURE("Error in DC group");
 
+  bool do_progressive = false;
+  if (pause_at_progressive_) {
+    do_progressive = true;
+    if (single_section) {
+      // If there's only one group and one pass, there is no separate section
+      // for DC and the entire full resolution image is available at once.
+      do_progressive = false;
+    }
+    if (!decoded_->metadata()->extra_channel_info.empty()) {
+      // If extra channels are encoded with modular without squeeze, they
+      // don't support DC. If the are encoded with squeeze, DC works in theory
+      // but the implementation may not yet correctly support this for Flush.
+      // Therefore, can't correctly pause for a progressive step if there is
+      // an extra channel (including alpha channel)
+      do_progressive = false;
```

Comment:
I'd add a TODO about checking whether this is still the case (@jonsneyers too)

Reply:
done

Reply:
I think https://github.com/libjxl/libjxl/pull/607 and https://github.com/libjxl/libjxl/pull/629 should make progressive alpha work (assuming it is squeezed of course, if not then there is only 1:1 group data, no progressive previews). 
Best to test if it actually still works.

The example application from https://github.com/libjxl/libjxl/pull/606 can be useful as a start to make a debug tool that dumps the image at every progression event (for the various settings of progressive detail). It dumps .pam files (not .ppm or .pfm) exactly to make it possible to see what is happening with the alpha channel too.